CMakeLists.txt

```diff
@@ -58,15 +61,13 @@ if(USE_CUDA AND CMAKE_CUDA_COMPILER)
     "${MODELS_ROOT}/*.cu"
     "${MODELS_ROOT}/*.cuh"
   )
-  list(APPEND generator_srcs ${generator_cuda_srcs})
+  list(APPEND generator_srcs ${generator_cuda_srcs} ${generator_cuda_exclude_srcs})
   add_compile_definitions(USE_CUDA=1)
   include_directories("${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}")
 elseif(USE_CUDA)
   # USE_CUDA is true but cmake could not find the cuda compiler
-  message(FATAL_ERROR "USE_CUDA is ON but no cuda compiler was found.")
-else()
-  file(GLOB generator_cuda_srcs "${GENERATORS_ROOT}/*_cuda*.*")
-  list(REMOVE_ITEM generator_srcs ${generator_cuda_srcs})
+  message(WARNING "USE_CUDA is ON but no cuda compiler was found. Disabling CUDA support.")
```

Comment:
What is the context behind giving a warning instead of terminating the build?

Reply:
Because USE_CUDA is ON by default. So if someone work on a machine doesn't have CUDA, they have to change USE_CUDA to OFF every time sync with main.

Reply:
They can turn it off using -DUSE_CUDA=OFF, right?
Silently falling back to not using cuda when the user requests it (for example when user says -DUSE_CUDA=ON) might give users the wrong impression of what is getting built.

Reply:
But I think by default, most of user doesn't have CUDA. And we don't tell them how to build without CUDA from source (-DUSE_CUDA=OFF).